lib/puppet/reports/http.rb

```diff
@@ -14,7 +14,7 @@
   def process
     url = URI.parse(Puppet[:reporturl])
     headers = { "Content-Type" => "application/x-yaml" }
-    options = {}
+    options = { :metric_id => [:puppet, :report, :http] }
```

Comment:
Ok I'm confused again :) It looks like at https://github.com/puppetlabs/puppet/blob/master/lib/puppet/network/http/connection.rb#L199-L205 the only option we propagate is `basic_auth` - nothing else is retained as the connection is executed, https://github.com/puppetlabs/puppet/blob/master/lib/puppet/network/http/connection.rb#L172. Does this use different connection logic somewhere? I'm afraid I'm not super familiar with the puppet server ruby http client and where it intersects with the puppet agent http client code.